src/app/search/page.tsx

```
@@ -76,13 +78,40 @@ function SearchBar({
+	const { showLoginToast, showToast } = useToast();
 
-	const handleBookmark = (e: React.MouseEvent) => {
+	// 로그인 상태에 따라 북마크 상태 결정
+	const [bookmarked, setBookmarked] = useState(() =>
+		user ? isScraped(item.id) : isBookmarked(item.id)
+	);
+
+	// user나 isScraped가 변경되면 상태 업데이트
+	useEffect(() => {
+		if (user) {
+			setBookmarked(isScraped(item.id));
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/src/app/search/page.tsx:92:4
  90 | 	useEffect(() => {
  91 | 		if (user) {
> 92 | 			setBookmarked(isScraped(item.id));
     | 			^^^^^^^^^^^^^ Avoid calling setState() directly within an effect
  93 | 		} else {
  94 | 			setBookmarked(isBookmarked(item.id));
  95 | 		}

(react-hooks/set-state-in-effect)